app/commands/git/sync_practice_exercise.rb

```diff
@@ -24,8 +24,6 @@ def call
     attr_reader :exercise
 
     def exercise_needs_updating?
-      return false if synced_to_head?
```

Comment:
This was not needed as the synced_to_head check was already done at the level of the track syncing (which calls this file's class)